### PR TITLE
Allow for setting the size of the kernel instead of a threshold in RelaxationBlock

### PR DESCRIPTION
**Pull request by masterleinad**
So far, we could only choose the singular values considered to be zero using a threshold.
This adds the possibility to set the desired kernel size on each block which is useful if we already know what to expect.

### CODE REVIEW

File: include/deal.II/lac/lapack_full_matrix.h [hunk 1]
```diff
@@ -746,6 +746,12 @@ class LAPACKFullMatrix : public TransposeTable<number>
    */
   void compute_inverse_svd (const double threshold = 0.);
 
+  /**
+   * Same as above but provide the size of the kernel instead of a threshold,
+   * i.e. the @p kernel_size smallest eigenvalues.
+   */
+  void compute_inverse_svd (const unsigned int kernel_size = 0);
```
Comment (tjhei) [inline include/deal.II/lac/lapack_full_matrix.h]: You can have two functions with the same name with default arguments? How does that work? Regardless, I don't think that is a good idea.

**Comment by tjhei** [comment]
I don't feel strongly about it, but I think it is confusing that
```
compute_inverse_svd();
compute_inverse_svd(0.);
compute_inverse_svd(0);
```
does different things.

**Comment by bangerth** [comment]
On 03/07/2018 03:53 PM, Timo Heister wrote:
> I don't feel strongly about it, but I think it is confusing that
> 
> |compute_inverse_svd(); compute_inverse_svd(0.); compute_inverse_svd(0); |
> 
> does different things.

Specifically, you could previously write
   compute_inverse_svd(0)
and call the function that takes a double argument. But you're now calling the 
other function, so the change is not backward compatible.

**Pull request merged**